File: Plugins~/Src/Loader/ImageData.cpp
#include "stdafx.h"

#include "ImageData.h"

namespace StreamingImageSequencePlugin {

ImageData::ImageData() 
    : RawData(NULL)
    , Width (0)
    , Height (0)
    , CurrentReadStatus(READ_STATUS_NONE)

{
}

//----------------------------------------------------------------------------------------------------------------------

ImageData::ImageData(uint8_t* _rawData, uint32_t _width, uint32_t _height, ReadStatus _readStatus) 
    : RawData(_rawData)
    , Width (_width)
    , Height (_height)
    , CurrentReadStatus(_readStatus)
{
}

} //end namespace

File: Runtime/Scripts/PluginAPI/ImageData.cs
using System;
using System.Runtime.InteropServices;

namespace Unity.StreamingImageSequence {
//----------------------------------------------------------------------------------------------------------------------

[StructLayout(LayoutKind.Sequential, CharSet = CharSet.Unicode, Pack = 8)]
internal struct ImageData {
    public IntPtr RawData;
    [MarshalAs(UnmanagedType.I4)]
    public readonly int Width;
    [MarshalAs(UnmanagedType.I4)]
    public readonly int Height;
    [MarshalAs(UnmanagedType.I4)]
    public readonly int ReadStatus;
    [MarshalAs(UnmanagedType.I4)]
    public readonly int Format;

    internal ImageData(int readStatus) {
        RawData    = IntPtr.Zero;
        Width      = 0;
        Height     = 0;        
        ReadStatus = readStatus;
        Format     = StreamingImageSequenceConstants.IMAGE_FORMAT_RGBA32;

    }
};


} // end namespace